### PR TITLE
refactor: migrate online archive resource and data sources to new SDK

### PR DESCRIPTION
**Pull request by AgustinBettati**
## Description

As part of [INTMDB-1110](https://jira.mongodb.org/browse/INTMDB-1110), this PR only contains changes to migrate online archive resource and data sources to new sdk. Followup PR will be made adding support for a new `dataExpirationRule` property.

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.

### CODE REVIEW

File: mongodbatlas/resource_mongodbatlas_online_archive.go [hunk 1]
```diff
@@ -564,38 +562,36 @@ func mapSchedule(d *schema.ResourceData) *matlas.OnlineArchiveSchedule {
...
-	if dayOfWeek, ok := scheduleTFConfig["day_of_week"].(int); ok {
-		scheduleInput.DayOfWeek = int32(dayOfWeek)
+	if dayOfWeek, ok := scheduleTFConfig["day_of_week"].(int); ok && dayOfWeek != 0 { // needed to verify attribute is actually defined
+		scheduleInput.DayOfWeek = pointy.Int(dayOfWeek)
 	}
 
-	if dayOfMonth, ok := scheduleTFConfig["day_of_month"].(int); ok {
-		scheduleInput.DayOfMonth = int32(dayOfMonth)
+	if dayOfMonth, ok := scheduleTFConfig["day_of_month"].(int); ok && dayOfMonth != 0 {
+		scheduleInput.DayOfMonth = pointy.Int(dayOfMonth)
```
Comment (AgustinBettati) [inline mongodbatlas/resource_mongodbatlas_online_archive.go:594]: only setting value when `!= 0` to preserve same behaviour. If this check is not added, property with default value 0 is always sent and causes errors in API.

**Review by andreaangiolillo — APPROVED** [review]
LGTM

File: mongodbatlas/data_source_mongodbatlas_online_archive.go [hunk 2]
```diff
@@ -187,48 +186,49 @@ func schemaOnlineArchive() map[string]*schema.Schema {
...
 
 	if err != nil {
 		return diag.FromErr(fmt.Errorf("error getting Online Archives list for project(%s) in cluster (%s): (%s)", projectID, clusterName, err.Error()))
 	}
 
 	results := make([]map[string]interface{}, 0, len(archives.Results))
 
-	for _, archive := range archives.Results {
-		archiveData := fromOnlineArchiveToMap(archive)
+	for i := range archives.Results {
```
Comment (lantoli) [inline mongodbatlas/data_source_mongodbatlas_online_archive.go:230]: nice performance trick

File: mongodbatlas/resource_mongodbatlas_online_archive.go [hunk 3]
```diff
@@ -306,44 +309,46 @@ func resourceMongoDBAtlasOnlineArchiveImportState(ctx context.Context, d *schema
...
 
-func mapToArchivePayload(d *schema.ResourceData) matlas.OnlineArchive {
+func mapToArchivePayload(d *schema.ResourceData) admin.BackupOnlineArchiveCreate {
 	// shared input
-	requestInput := matlas.OnlineArchive{
-		DBName:         d.Get("db_name").(string),
-		CollName:       d.Get("coll_name").(string),
-		CollectionType: d.Get("collection_type").(string),
+	requestInput := admin.BackupOnlineArchiveCreate{
+		DbName:   d.Get("db_name").(string),
```
Comment (lantoli) [inline mongodbatlas/resource_mongodbatlas_online_archive.go:347]: type conversions are normally recommended with error var

**Review by lantoli — APPROVED** [review]

File: mongodbatlas/resource_mongodbatlas_online_archive.go (hunk 3 above)
Comment (AgustinBettati) [reply mongodbatlas/resource_mongodbatlas_online_archive.go:347]: it is a valid point, I would however consider that value is obtained from terraform where resource schema is being enforced.

**Pull request merged**